repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingContextMap.scala

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.httplogging
+
+import java.time.Duration
+import java.util
+
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import org.openrepose.commons.utils.http.CommonHttpHeader
+import org.openrepose.commons.utils.logging.TracingHeaderHelper
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+import scala.util.Try
+
+// TODO: This utility is not strictly necessary. Since JTwig allows access to
+// TODO: Java objects via the selection operator, we could pass the context
+// TODO: object itself into the template. That would help remove some redundancy
+// TODO: but may or may not improve user experience.
+/**
+  * Converts an [[HttpLoggingContext]] into a [[Map]] for use by the templating
+  * engine.
+  */
+object HttpLoggingContextMap {
+
+  private final val Generators: Map[String, HttpLoggingContext => Any] = Map(
+    "inboundRequestProtocol" -> (_.getInboundRequest.getProtocol),
+    "outboundRequestProtocol" -> (_.getOutboundRequest.getProtocol),
+    "inboundResponseProtocol" -> (_.getInboundResponseProtocol),
+    "inboundRequestMethod" -> (_.getInboundRequest.getMethod),
+    "outboundRequestMethod" -> (_.getOutboundRequest.getMethod),
+    "inboundRequestPath" -> (_.getInboundRequest.getRequestURI),
+    "outboundRequestPath" -> (_.getOutboundRequest.getRequestURI),
+    "inboundRequestQueryString" -> (_.getInboundRequest.getQueryString),
+    "outboundRequestQueryString" -> (_.getOutboundRequest.getQueryString),
+    "inboundRequestUrl" -> (_.getInboundRequest.getRequestURL.toString),
+    "outboundRequestUrl" -> (_.getOutboundRequest.getRequestURL.toString),
+    "inboundRequestHeaders" -> (ctx => headerMap(ctx.getInboundRequest)),
+    "outboundRequestHeaders" -> (ctx => headerMap(ctx.getOutboundRequest)),
+    "outboundResponseHeaders" -> (ctx => headerMap(ctx.getOutboundResponse)),
+    "outboundResponseStatusCode" -> (_.getOutboundResponseStatusCode),
+    "outboundResponseReasonPhrase" -> (_.getOutboundResponseReasonPhrase),
+    "outboundResponseContentLength" -> (_.getOutboundResponseContentLength),
+    "outboundResponseBytesWritten" -> (_.getOutboundResponseBytesWritten),
+    "timeRequestReceived" -> (_.getTimeRequestReceived),
+    "timeToHandleRequest" -> (ctx => Duration.between(ctx.getTimeRequestReceived, ctx.getTimeRequestCompleted)),
+    "timeInOriginService" -> (_.getTimeInOriginService),
+    "localIpAddress" -> (_.getInboundRequest.getLocalAddr),
+    "remoteIpAddress" -> (_.getInboundRequest.getRemoteAddr),
+    "remoteHost" -> (_.getInboundRequest.getRemoteHost),
+    "traceId" -> (ctx => TracingHeaderHelper.getTraceGuid(ctx.getOutboundRequest.getHeader(CommonHttpHeader.TRACE_GUID))),
+    "extensions" -> (_.getExtensions)
```

Comment:
I'm just pointing out that these names are not single-character keys. I am personally a fan of the more descriptive keys, especially since JTwig seems built to use them (i.e., expressions are fully enclosed `{{ example }}` rather than being prefixed `%I`).

Reply:
I concur